src/Microsoft.Health.Dicom.Core.UnitTests/Features/Retrieve/RetrieveMetadataHandlerTests.cs

```diff
@@ -89,6 +94,7 @@ public async Task GivenARequestWithInvalidStudyAndSeriesInstanceIdentifier_WhenR
         [InlineData("aaaa-bbbb2", " ")]
         public async Task GivenARequestWithInvalidSeriesInstanceIdentifier_WhenRetrievingInstanceMetadata_ThenDicomInvalidIdentifierExceptionIsThrown(string seriesInstanceUid, string sopInstanceUid)
         {
+            EnsureArg.IsNotNull(seriesInstanceUid, nameof(seriesInstanceUid));
             string ifNoneMatch = null;
```

Comment:
Check `SopInstanceUid`?

Reply:
Again same as above, only checking minimal necessary. This function actually calls .trim . Also this is tests I tend to allow as much as possible to facilitate getting results from the underlying code.